lib/pipeline/RequestHandler.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const Duplex  = require('readable-stream').Duplex;
+const metrics = require('auth0-instrumentation').metrics;
+
+class RequestHandler extends Duplex {
+  constructor(options) {
+    super({
+      writableObjectMode: true,
+      readableObjectMode: true,
+      highWaterMark: 1000
+    });
+
+    this.logger = options.logger;
+    this.db = options.db;
+    this.queue = [];
+  }
+
+  _commonReponseHandler(request, callback) {
+    const start = new Date();
+    return (err, result) => {
+      const took = new Date() - start;
+
+      metrics.histogram('limitd.database.time', (new Date() - start), {
+        method: request.method,
+        type: request.type
+      });
+
+      if (err) {
+        if (err.message.indexOf('undefined bucket type') > -1 && !request.skipResponse) {
+          this._queueResponse(request, { error: { type: 'UNKNOWN_BUCKET_TYPE' } });
+        } else {
+          this.emit('error', err);
+        }
+        return callback();
+      }
+
+      this.logger.debug('finished operation', {
+        took:       took,
+        method:     request.method,
+        type:       request.type,
+        conformant: result.conformant
+      });
+
+      if (!request.skipResponse) {
+        this._queueResponse(request, result);
+      }
+      callback();
+    };
+  }
+
+  _write(request, encoding, callback) {
+    switch(request.method) {
```

Comment:
so this is the new router? is it easy to find? should we make it explicit? I think Handler is OK, but this is both Router and Handler